src/DynamoCore/Nodes/CodeBlockNode.cs

```diff
@@ -310,14 +311,25 @@ internal override IEnumerable<AssociativeNode> BuildAst(List<AssociativeNode> in
 
             foreach (var astNode in codeStatements.Select(stmnt => NodeUtils.Clone(stmnt.AstNode)))
             {
-                MapIdentifiers(astNode);
+                if (context != AstBuilder.CompilationContext.NodeToCode)
+                    MapIdentifiers(astNode);
                 resultNodes.Add(astNode as AssociativeNode);
             }
 
             return resultNodes;
         }
 
-        public override IdentifierNode GetAstIdentifierForOutputIndex(int portIndex)
+        /// <summary>
+        /// For code block node, the identifer for its outupt is mapped. For 
+        /// example, the output identifier of expression "x = 1" is 
+        /// "x" + CBN's GUID. This mapping can ensure the uniqueness of variable
+        /// name. But in some case, we need to get its original name instead of
+        /// mapped one.
+        /// </summary>
+        /// <param name="portIndex">Output port index</param>
+        /// <param name="forRawName">Indicate to get original name</param>
+        /// <returns></returns>
+        private IdentifierNode GetAstIdentifierForOutputIndexInternal(int portIndex, bool forRawName)
```

Comment:
As clear as daylight, thanks a lot! Here's how I'd suggest a rephrase, if you don't mind:

```csharp
/// <summary>
/// For code block nodes, each output identifier of an output port is mapped.
/// For an example, "p = 1" would have its internal identifier renamed to 
/// "pXXXX", where "XXXX" is the GUID of the code block node. This mapping is 
/// done to ensure the uniqueness of the output variable name.
/// </summary>
/// <param name="portIndex">Output port index</param>
/// <param name="forRawName">Set this parameter to true to retrieve the 
/// original identifier name "p". If this parameter is false, the mapped 
/// identifer name "pXXXX" is returned instead.</param>
/// <returns></returns>
```